sbin/start-history-server.sh

```diff
@@ -34,6 +34,7 @@ if [ $# != 0 ]; then
   echo "Using command line arguments for setting the log directory is deprecated. Please "
   echo "set the spark.history.fs.logDirectory configuration option instead."
   export SPARK_HISTORY_OPTS="$SPARK_HISTORY_OPTS -Dspark.history.fs.logDirectory=$1"
+  all_other_args="${@:2}"
```

Comment:
This seems sub optimal, the only way to specify a different properties file involves specifying a different log directory, and in a deprecated fashion. I think putting the argument parsing logic in HistoryServerArguments might be better (option #3 from the JIRA) but thats just my opinion, maybe @andrewor14 might have a better idea as the original author of this file.